### ISSUE TITLE
prange leads to different results when slicing operators are used

### ISSUE THREAD

**Issue by sebbelese** [numba/numba#2646]
The following code illustrates 3 different functions that allocate a 3d array and fill it with indices. When slicing is used in the destination array index, the array is not updated as it should be.

The following code:

```
from numba import njit, prange
import numpy as np

@njit
def arrayFilledWithIdxSerial(size1, size2a, size2b):
    tab = np.zeros((size1,size2a*size2b, 2))
    for iT in range(size2a):
       for iX in range(size2b):
           for iP in range(size1):
                tab[iP,iT*size2b+iX,:] = np.array([iT, iX])
    return tab

@njit(parallel=True)
def arrayFilledWithIdxParallel(size1, size2a, size2b):
    tab = np.zeros((size1,size2a*size2b, 2))
    for iT in prange(size2a):
       for iX in range(size2b):
           for iP in range(size1):
                tab[iP,iT*size2b+iX,:] = np.array([iT, iX])
    return tab

@njit(parallel=True)
def arrayFilledWithIdxParallel2Steps(size1, size2a, size2b):
    tab = np.zeros((size1,size2a*size2b, 2))
    for iT in prange(size2a):
       for iX in range(size2b):
           for iP in range(size1):
                tab[iP,iT*size2b+iX,0] = iT
                tab[iP,iT*size2b+iX,1] = iX
    return tab

tabSer = arrayFilledWithIdxSerial(3,2,2)
print("Serial",tabSer)

tabPar = arrayFilledWithIdxParallel(3,2,2)
print("Parallel",tabPar)

tabPar = arrayFilledWithIdxParallel2Steps(3,2,2)
print("Parallel 2 steps",tabPar)
```

produces the following output:

```
Serial [[[ 0.  0.]
  [ 0.  1.]
  [ 1.  0.]
  [ 1.  1.]]

 [[ 0.  0.]
  [ 0.  1.]
  [ 1.  0.]
  [ 1.  1.]]

 [[ 0.  0.]
  [ 0.  1.]
  [ 1.  0.]
  [ 1.  1.]]]
Parallel [[[ 0.  0.]
  [ 0.  0.]
  [ 0.  0.]
  [ 0.  0.]]

 [[ 0.  0.]
  [ 0.  0.]
  [ 0.  0.]
  [ 0.  0.]]

 [[ 0.  0.]
  [ 0.  0.]
  [ 0.  0.]
  [ 0.  0.]]]
Parallel 2 steps [[[ 0.  0.]
  [ 0.  1.]
  [ 1.  0.]
  [ 1.  1.]]

 [[ 0.  0.]
  [ 0.  1.]
  [ 1.  0.]
  [ 1.  1.]]

 [[ 0.  0.]
  [ 0.  1.]
  [ 1.  0.]
  [ 1.  1.]]]
```

I don't know which are the exact conditions for this bug to appear.

Numba 0.36.1 on Windows (Miniconda)

**Comment by sebbelese** [comment]
Did anyone tried to reproduce this?

**Comment by ehsantn** [comment]
@sebbelese Looks like this issue is resolved on master with the recent ParallelAccelerator PR.

**Comment by sebbelese** [comment]
I can confirm that it is fixed on my current version of numba (0.37.0).
